pylint/checkers/refactoring.py

```diff
@@ -261,12 +265,30 @@ def _check_redefined_argument_from_local(self, name_node):
                                  args=(name_node.name, ))
 
     @utils.check_messages('redefined-argument-from-local',
-                          'too-many-nested-blocks')
+                          'too-many-nested-blocks',
+                          'mutable-sequence-modified-in-loop')
     def visit_for(self, node):
         self._check_nested_blocks(node)
 
         for name in node.target.nodes_of_class(astroid.AssignName):
             self._check_redefined_argument_from_local(name)
+        for part in node.body:
+            if isinstance(node.iter, astroid.node_classes.Name):
+                self._check_mutable_sequence_modified_in_loop(part,
+                                                              node.iter.name)
+
+    def _check_mutable_sequence_modified_in_loop(self, node, target_name):
```

Comment:
this method looks like it can be simplified by using `IF_BODY.nodes_of_class(astroid.Expr)`. It'll hide "recursive" nature of performed search.